[crossguid:x64-osx] build failure

**Host Environment**
- OS: [MacOS Catalina]
- Compiler: clang

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install crossguid`

**Failure logs** 
```
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:145 (message):
    Command failed: /Users/developer/Program/vcpkg/downloads/tools/cmake-3.19.2-osx/cmake-3.19.2-macos-universal/CMake.app/Contents/bin/cmake --build . --config Debug --target install -- -v -j17
    Working Directory: /Users/developer/Program/vcpkg/buildtrees/crossguid/x64-osx-dbg
    See logs for more information:
      /Users/developer/Program/vcpkg/buildtrees/crossguid/install-x64-osx-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:96 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:27 (vcpkg_build_cmake)
  ports/crossguid/portfile.cmake:18 (vcpkg_install_cmake)
  scripts/ports.cmake:136 (include)
```
install-x64-osx-dbg-out.log
```
[1/3] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DGUID_CFUUID -I/Users/developer/Program/vcpkg/buildtrees/crossguid/src/365d430337-f320045b03.clean/include -fPIC -g -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -Werror -Wall -std=c++11 -MD -MT CMakeFiles/crossguid.dir/src/guid.cpp.o -MF CMakeFiles/crossguid.dir/src/guid.cpp.o.d -o CMakeFiles/crossguid.dir/src/guid.cpp.o -c /Users/developer/Program/vcpkg/buildtrees/crossguid/src/365d430337-f320045b03.clean/src/guid.cpp
FAILED: CMakeFiles/crossguid.dir/src/guid.cpp.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DGUID_CFUUID -I/Users/developer/Program/vcpkg/buildtrees/crossguid/src/365d430337-f320045b03.clean/include -fPIC -g -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -Werror -Wall -std=c++11 -MD -MT CMakeFiles/crossguid.dir/src/guid.cpp.o -MF CMakeFiles/crossguid.dir/src/guid.cpp.o.d -o CMakeFiles/crossguid.dir/src/guid.cpp.o -c /Users/developer/Program/vcpkg/buildtrees/crossguid/src/365d430337-f320045b03.clean/src/guid.cpp
/Users/developer/Program/vcpkg/buildtrees/crossguid/src/365d430337-f320045b03.clean/src/guid.cpp:403:7: error: unused function 'swap' [-Werror,-Wunused-function]
        void swap(xg::Guid &lhs, xg::Guid &rhs)
             ^
1 error generated.
ninja: build stopped: subcommand failed.
```

--- a/ports/crossguid/portfile.cmake
+++ b/ports/crossguid/portfile.cmake
@@ -6,6 +6,8 @@ vcpkg_from_github(
     REF ca1bf4b810e2d188d04cb6286f957008ee1b7681 #2021-10-22
     SHA512 f0a80d8e99b10473bcfdfde3d1c5fd7b766959819f0d1c0595ac84ce46db9007a5fbfde9a55aca60530c46cb7f8ef4c7e472c6191559ded92f868589c141ccaf
     HEAD_REF master
+    PATCHES
+        warnings.patch
 )
 
 vcpkg_cmake_configure(

--- a/ports/crossguid/vcpkg.json
+++ b/ports/crossguid/vcpkg.json
@@ -1,12 +1,12 @@
 {
   "name": "crossguid",
   "version-date": "2021-10-22",
-  "port-version": 1,
+  "port-version": 2,
   "description": "CrossGuid is a minimal, cross platform, C++ GUID library.",
   "dependencies": [
     {
       "name": "libuuid",
-      "platform": "!windows & !uwp & !osx & !android"
+      "platform": "!windows & !osx & !android"
     },
     {
       "name": "vcpkg-cmake",

--- a/ports/crossguid/warnings.patch
+++ b/ports/crossguid/warnings.patch
@@ -1,0 +1,20 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 76b5a62..174d981 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -40,15 +40,6 @@ else()
+     target_compile_definitions(crossguid PRIVATE GUID_LIBUUID)
+ endif()
+ 
+-if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+-    set(WARNINGS "-Werror" "-Wall")
+-elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+-    set(WARNINGS "-Werror" "-Wall")
+-elseif(MSVC)
+-    set(WARNINGS "/WX" "/W4")
+-endif()
+-target_compile_options(crossguid PRIVATE ${WARNINGS})
+-
+ set_target_properties(crossguid
+ 					  PROPERTIES
+ 					  VERSION ${PROJECT_VERSION}

--- a/scripts/ci.baseline.txt
+++ b/scripts/ci.baseline.txt
@@ -176,7 +176,6 @@ crashpad:arm-uwp=fail
 crashpad:x64-linux=fail
 crashpad:x64-uwp=fail
 crashpad:x86-windows=fail
-crossguid:x64-osx=fail
 ctemplate:arm64-windows=fail
 ctemplate:arm-uwp=fail
 ctemplate:x64-linux=fail

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -1658,7 +1658,7 @@
     },
     "crossguid": {
       "baseline": "2021-10-22",
-      "port-version": 1
+      "port-version": 2
     },
     "crow": {
       "baseline": "0.3.4",

--- a/versions/c-/crossguid.json
+++ b/versions/c-/crossguid.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "93714099ddf83b2f3437a080c8acc08ff74cbe37",
+      "version-date": "2021-10-22",
+      "port-version": 2
+    },
+    {
       "git-tree": "4e53633ff35a2bcfb9076aee3e0d740406e788d3",
       "version-date": "2021-10-22",
       "port-version": 1

